presto-main/src/main/java/com/facebook/presto/operator/spiller/BinarySpillerFactory.java

```diff
@@ -33,28 +33,27 @@
 public class BinarySpillerFactory
         implements SpillerFactory
 {
-    public static final Path SPILL_PATH = Paths.get("/tmp/spills");
-
     private final ListeningExecutorService executor;
     private final BlockEncodingSerde blockEncodingSerde;
+    private final Path spillPath;
 
     @Inject
-    public BinarySpillerFactory(BlockEncodingSerde blockEncodingSerde)
+    public BinarySpillerFactory(BlockEncodingSerde blockEncodingSerde, FeaturesConfig featuresConfig)
     {
-        this(blockEncodingSerde, MoreExecutors.listeningDecorator(newFixedThreadPool(4, daemonThreadsNamed("binary-spiller-%s"))));
+        this(blockEncodingSerde, MoreExecutors.listeningDecorator(newFixedThreadPool(4, daemonThreadsNamed("binary-spiller-%s"))), featuresConfig);
     }
 
-    public BinarySpillerFactory(BlockEncodingSerde blockEncodingSerde, ListeningExecutorService executor)
+    public BinarySpillerFactory(BlockEncodingSerde blockEncodingSerde, ListeningExecutorService executor, FeaturesConfig featuresConfig)
     {
         this.blockEncodingSerde = requireNonNull(blockEncodingSerde, "blockEncodingSerde is null");
         this.executor = requireNonNull(executor, "executor is null");
-
-        SPILL_PATH.toFile().mkdirs();
+        this.spillPath = featuresConfig.getSpillerSpillPath();
+        this.spillPath.toFile().mkdirs();
```

Comment:
what's wrong with that? When someone configures spilling to be enabled this directory for 99% would already exists anyway. If not, if spilling is enabled I expect that it's not a mistake and query will indeed spill, so it will be used anyway. So what's the point in complicating the code by moving it to `BinaryFileSpiller`?